src/pages/DelegationCreatePage/components/DelegationCreateForm.tsx

```diff
@@ -0,0 +1,262 @@
+import { RHFAutocompleteSingle, RHFTextField } from '@/components/shared/react-hook-form-inputs'
+import { Box, FormControlLabel, Switch } from '@mui/material'
+import React, { useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+import { StepActions } from '@/components/shared/StepActions'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import SendIcon from '@mui/icons-material/Send'
+import { EServiceQueries } from '@/api/eservice/eservice.queries'
+import { useQuery } from '@tanstack/react-query'
+import {
+  CompactOrganization,
+  EServiceMode,
+  EServiceTechnology,
+  ProducerEService,
+} from '@/api/api.generatedTypes'
+import { useAutocompleteTextInput } from '@pagopa/interop-fe-commons'
+import { SectionContainer } from '@/components/layout/containers'
+import { useDialog } from '@/stores'
+import { TenantQueries } from '@/api/tenant'
+
+export type DelegationCreateFormValues = {
+  eserviceName: string
+  eserviceDescription: string
+  delegateId: string
+}
+
+export type EServiceCreateDraftValues = {
+  name: string
+  description: string
+  technology: EServiceTechnology
+  mode: EServiceMode
+}
+
+type DelegationCreateFormProps = {
+  delegationKind: 'CONSUME' | 'PROVIDE'
+  setActiveStep: React.Dispatch<React.SetStateAction<'KIND' | 'FORM'>>
+}
+
+const defaultValues: DelegationCreateFormValues = {
+  eserviceName: '',
+  eserviceDescription: '',
+  delegateId: '',
+}
+
+export const DelegationCreateForm: React.FC<DelegationCreateFormProps> = ({
+  delegationKind,
+  setActiveStep,
+}) => {
+  const { t } = useTranslation('party')
+
+  const [isChecked, setIsChecked] = useState(false)
+
+  const { openDialog } = useDialog()
+
+  const formMethods = useForm({ defaultValues })
+
+  const onSubmit = async (formValues: DelegationCreateFormValues) => {
+    if (!isChecked && delegationKind === 'PROVIDE') {
+      // if it is a producer delegation and isChecked is false the eservice must be created
+      const eserviceParams: EServiceCreateDraftValues = {
+        name: formValues.eserviceName,
+        description: formValues.eserviceDescription,
+        technology: 'REST',
+        mode: 'DELIVER',
+      }
+      openDialog({
+        type: 'delegations',
+        eserviceParams: eserviceParams,
+        delegationParams: { delegateId: formValues.delegateId },
+      })
```

Comment:
As I said before, I would prefer to have the submit logic here and not inside the dialog